mobile/src/people/people-company.tsx

```diff
@@ -1,23 +1,95 @@
-import { View } from 'react-native';
+import { connect, Dispatch, MapStateToProps, MapDispatchToPropsFunction } from 'react-redux';
+import { View, Text, ScrollView, Dimensions, Animated } from 'react-native';
 
 import { EmployeesList } from './employees-list';
+import { Department } from '../reducers/organization/department.model';
 import { AppState } from '../reducers/app.reducer';
+import { DepartmentsHScrollableList } from './departments/departments-horizontal-scrollable-list';
+import { DepartmentsTree } from './departments/departments-tree';
+import { DepartmentsTreeNode } from './departments/departments-tree-node';
+import { EmployeeCardWithAvatar } from './employee-card-with-avatar';
+import { PeopleActions, updateDepartmentIdsTree } from '../reducers/people/people.action';
+import { loadEmployeesForDepartment } from '../reducers/organization/organization.action';
 
 interface PeopleCompanyProps {
     routeName: string;
+    headDepartment: Department;
+    departmentsTree: DepartmentsTree;
+    departmentsBranch?: DepartmentsTreeNode[];
 }
 
 const mapStateToProps = (state: AppState): PeopleCompanyProps => ({
-    routeName: 'Company'
+    routeName: 'Company',
+    headDepartment: state.people.headDepartment,
+    departmentsTree: state.people.departmentsTree,
+    departmentsBranch: state.people.departmentsBranch
 });
 
-export class PeopleCompanyImpl extends React.Component<PeopleCompanyProps> {  
+interface PeopleCompanyDispatchProps {
+    requestEmployeesForDepartment: (departmentId: string) => void;
+    updateDepartmentIdsTree: (index: number, department: DepartmentsTreeNode) => void;
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<PeopleActions>) => ({
+    requestEmployeesForDepartment: (departmentId: string) => { 
+        dispatch(loadEmployeesForDepartment(departmentId)); 
+    },
+    updateDepartmentIdsTree: (index: number, department: DepartmentsTreeNode) => { 
+        dispatch(updateDepartmentIdsTree(index, department)); 
+    },
+});
+
+export class PeopleCompanyImpl extends React.Component<PeopleCompanyProps & PeopleCompanyDispatchProps> {
+    public treeRecurseAndAdd(department: DepartmentsTreeNode, departments: DepartmentsTreeNode[]) {
+        departments.push(department);
+        var children = department.children;
+        if (children !== null) {
+            children.forEach(child => this.treeRecurseAndAdd(child, departments));
+        }
+    }
+
     public render() {
-        return <View style={{flex: 1, backgroundColor: '#fff'}}/>;
+        const { children } = this.props.departmentsTree.root;
+
+        if (this.props.departmentsBranch === null) {
+            return <ScrollView style={{ backgroundColor: '#fff' }} />;
+        }
+
+        var heads: DepartmentsTreeNode[] = [];
+        var flattenDepartmentsNodes: DepartmentsTreeNode[] = [];
+        this.treeRecurseAndAdd(this.props.departmentsTree.root, flattenDepartmentsNodes);
+
+        this.props.departmentsBranch.map((department) => {
```

Comment:
do not use map for changing foreign variables, it's counter-intuitive (map is a pure function that maps one type to another and can even be lazy) and hard to debug. Use `for (const department of departmentsBranch){}` loop instead